angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
```js
angular.injector(['ng']).get('orderByFilter')('A', x => x.toLowerCase()); // ["a"]
```
Whatever is passed to `orderByFilter`, is coerced wrapped into a singleton array if it is not an array before continuing.

I'm starting to thing the whole premise of the filter is unsound.